.github/workflows/generate-dsl-info.yaml

```diff
@@ -0,0 +1,43 @@
+name: Generate DSL Info
+
+permissions:
+  contents: write
+
+on:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+jobs:
+  generate-dsl-info:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+
+      - name: Install dependencies
+        run: npm ci
```

Comment:
This repo uses pnpm. An example of setup: https://github.com/RSS3-Network/DSL-js-sdk/blob/f8ac11e5e64e101ac70b9694f326a8811e5a67bd/.github/workflows/release.yml#L18-L33